core/src/HIP/Kokkos_HIP.cpp

```diff
@@ -105,18 +104,10 @@ Kokkos::HIP::initialize WARNING: Could not determine that xnack is enabled.
   // Init the array for used for arbitrarily sized atomics
   desul::Impl::init_lock_arrays();  // FIXME
 
-  // Allocate a staging buffer for constant mem in pinned host memory
-  // and an event to avoid overwriting driver for previous kernel launches
-
-  void* constant_mem_void_ptr = nullptr;
-  KOKKOS_IMPL_HIP_SAFE_CALL(hipHostMalloc(
-      &constant_mem_void_ptr, Impl::HIPTraits::ConstantMemoryUsage));
-  Impl::HIPInternal::constantMemHostStaging =
-      static_cast<unsigned long*>(constant_mem_void_ptr);
-
-  KOKKOS_IMPL_HIP_SAFE_CALL(
-      hipEventCreate(&Impl::HIPInternal::constantMemReusable));
+  // Set singleton device id
```

Comment:
Not really a singleton, but maybe we can fix that naming issue in a separate PR later (also for CUDA etc.)